torch/cuda/streams.py

```diff
@@ -112,6 +112,29 @@ def __repr__(self):
                 .format(self.device, self.cuda_stream))
 
 
+class ExternalStream(Stream):
+    r"""Wrapper around a externally allocated CUDA stream.
+
+    A CUDA stream is a linear sequence of execution that belongs to a specific
+    device, independent from other streams.  See :ref:`cuda-semantics` for
+    details.
+
+    This class is used to wrap streams allocated in other libraries in order
+    to facilitate data exchange and multi-library interactions. Be aware
+    that the stream life-cycle is managed by the owner.
```

Comment:
This is a nice explicit mention of the stream's life cycle, but I think we might want to be even clearer here. 

Maybe this could explain that the ExternalStream object doesn't keep the underlying stream alive, and that the user is expected to keep the original stream alive as long as the external stream object is in scope / alive?

Reply:
Changed to the following paragrah

```
   This class doesn't manage the stream life-cycle, it is the user
    responsibility to keep the referenced stream alive while this class is
    being used.
```